src/syntree.rs

```diff
@@ -189,6 +189,115 @@ pub struct Type {
     kind: TypeKind,
 }
 
-pub fn construct(tokens: TokenStream) -> Result<Prog, Vec<Error>> {
-    Err(Vec::new())
+type Tokens = [(Token, usize)];
+
+#[derive(Debug, Copy, Clone)]
+struct Context<'a> {
+    pub tokens: &'a Tokens,
+    pub curr: usize,
+    pub file: &'a Path,
+}
```

Comment:
Unsure what you mean. I'm guessing you wonder why I made it a borrow instead of moving in the `PathBuf` into the struct? I wanted to make `Context` copy if I could, and since all of it is pointers that don't own any of the memory, a copy is perfectly legal. This is nice since we clone it alot when parsing, so we always can revert back.